src/app/routes/getInitialData/index.js

```diff
@@ -6,19 +6,48 @@ import onClient from '#lib/utilities/onClient';
 import getBaseUrl from './utils/getBaseUrl';
 import getPreprocessorRules from './utils/getPreprocessorRules';
 
-const logger = nodeLogger(__filename);
+export const logger = nodeLogger(__filename);
 const STATUS_CODE_OK = 200;
 const STATUS_CODE_BAD_GATEWAY = 502;
 const STATUS_CODE_NOT_FOUND = 404;
 const upstreamStatusCodesToPropagate = [STATUS_CODE_OK, STATUS_CODE_NOT_FOUND];
+const validRendererEnvironments = ['int', 'test', 'live'];
 
 const ampRegex = /(.amp)$/;
 
 const baseUrl = onClient()
   ? getBaseUrl(window.location.origin)
   : process.env.SIMORGH_BASE_URL;
 
-const getUrl = pathname => `${baseUrl}${pathname.replace(ampRegex, '')}.json`;
+// eslint-disable-next-line consistent-return
+export const getUrl = pathname => {
+  if (pathname) {
+    const regex = /renderer_env=(.*)&?/i;
+    const [match, environment] = pathname.match(regex) || [];
+    let finalPath = pathname;
+    let params =
+      pathname.lastIndexOf('?') > 0
+        ? pathname.substring(pathname.lastIndexOf('?'))
+        : '';
+
+    if (params) {
+      finalPath = pathname.substring(0, pathname.indexOf(params));
+
+      if (match) {
+        params = params.replace(match, match.toLowerCase());
+        if (!validRendererEnvironments.includes(environment.toLowerCase())) {
+          logger.warn(
+            `Invalid parameter value [${match}]. Usage: renderer_env=${validRendererEnvironments.join(
+              '|',
+            )}`,
+          );
+        }
+      }
+    }
```

Comment:
bit more going on here than I thought - wasn't expecting this to effectively allow support for any querystring parameter at all.  Seems okay to do, but I'd recommend addressing the code climate warning re: complexity if we are doing this.  It seems a legitimate warning in this case - there's a lot going on here

eg, line 38-44 being something like `validateRendererEnvironment(environment)` would remove some complexity - other suggestions would be things like `getQuerystring(pathname)` and `getRendererEnvironment(querystring)`

Reply:
Good shout - there was definitely a lot going on.  I have now made this much simpler.